code/mob/melee_attack_procs.dm

```diff
@@ -34,13 +34,21 @@
 	M.update_burning(-1.2)
 	playsound(M.loc, 'sound/impact_sounds/Generic_Shove_1.ogg', 50, 1, 0 , 0.7)
 	src.visible_message("<span class='notice'>[src] pats down [M] wildly, trying to put out the fire!</span>")
+	var/glove_protection = FALSE
 
-	if (prob(50))
+	if (ishuman(src)) // glove heat resistance check
+		var/mob/living/carbon/human/H = src
+		var/obj/item/clothing/gloves/G = H.gloves
+		if (G && G.hasProperty("heatprot"))
+			if (G.getProperty("heatprot") >= 7)
+				glove_protection = 1
```

Comment:
Why use a bool if you can put the stuff below this here instead?

Reply:
Changed that a commit